src/main/java/org/jitsi/stats/DataDogStatsClient.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright @ 2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.stats;
+
+import com.timgroup.statsd.*;
+import org.jitsi.util.*;
+
+/**
+ * Encapsulates a {@link NonBlockingStatsDClient} to be able to send
+ * statistics to local data-hog server
+ *
+ * @author Nik Vaessen
+ */
+public class DataDogStatsClient
+{
+    /**
+     * The logger of this class
+     */
+    private static final Logger logger
+        = Logger.getLogger(DataDogStatsClient.class);
+
+    /**
+     * Single instance of this {@link DataDogStatsClient}
+     */
+    private static DataDogStatsClient singleton = new DataDogStatsClient();
```

Comment:
do we need to enforce the singleton here?  can we not let the calling code decide how it wants to manage the lifetime of the client?

Reply:
Can we follow the convention used by other things exposed by jicoco and make this an OSGi service?

Reply:
i'd like to not force the usage of OSGi--we can set it up so that the class could be used directly and provide an OSGi wrapper for those who want to use that.